spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/filters/post/SendErrorFilter.java

```diff
@@ -76,6 +77,8 @@ public Object run() {
 				request.setAttribute("javax.servlet.error.message", message);
 			}
 
+			//unwrap request if needed, otherwise tomcat hangs in endless loop
```

Comment:
Even after removing the send 404 to `/error` I still need to unwrap. I'm a bit concerned and wonder if anyone can comment.

The problem occurs in [`ApplicationDispatcher`]( https://github.com/apache/tomcat/blob/TOMCAT_8_0_0/java/org/apache/catalina/core/ApplicationDispatcher.java#L907-L908).

```
if (!(current instanceof ServletRequestWrapper))
                break;
```
The request is always a `ServletRequestWrapper` unless I unwrap it.

@markt-asf @wilkinsona @philwebb @snicoll any thoughts?